### PR TITLE
Random forest

### PR DESCRIPTION
**Pull request by j3r3m1**
Add the first IProcess to create the randomForest from a training dataset in a ITable.
It is not yet the final version since for the moment there is not file to download on the internet (I have just a local version).

### CODE REVIEW

File: geoindicators/pom.xml [hunk 1]
```diff
@@ -45,6 +45,18 @@
             <artifactId>h2gis-network</artifactId>
             <version>2.0.0-SNAPSHOT</version>
         </dependency>
+        <dependency>
+            <groupId>com.github.haifengl</groupId>
+            <artifactId>smile-core</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>com.github.haifengl</groupId>
+            <artifactId>smile-netlib</artifactId>
```
Comment (ebocher) [inline geoindicators/pom.xml]: Is this dep necessary ?

File: geoindicators/src/main/groovy/org/orbisgis/orbisprocess/geoclimate/geoindicators/TypologyClassification.groovy [hunk 2]
```diff
@@ -240,3 +248,75 @@ IProcess identifyLczType() {
...
+
+            // Calculate the prediction using the same sample in order to identify what is the
+            // data rate that has been well classified
+            int[] prediction = Validation.test(model, df)
+            int[] truth = df.apply(varToModel).toIntArray()
+            double accuracy = Accuracy.of(truth, prediction)
+            logger.info "The percentage of the data that have been well classified is : ${accuracy*100}%"
+
+            if (save){
+                XStream xs = new XStream()
```
Comment (ebocher) [inline geoindicators/src/main/groovy/org/orbisgis/orbisprocess/geoclimate/geoindicators/TypologyClassification.groovy]: Perhaps use jackson instead of XStream because it's a new dep.
@SPalominos

File: geoindicators/src/main/groovy/org/orbisgis/orbisprocess/geoclimate/geoindicators/DataUtils.groovy [hunk 3]
```diff
@@ -34,14 +35,27 @@ IProcess joinTables() {
             inputTableNamesWithId.each { key, value ->
                 if (alias == "a") {
                     columnKey = "$alias.${value}"
-                    columns = datasource.getTable(key).columnNames.collect {
-                        alias + "." + it
```
Comment (ebocher) [inline geoindicators/src/main/groovy/org/orbisgis/orbisprocess/geoclimate/geoindicators/DataUtils.groovy]: This add-on must be set in a separate PR.

File: geoindicators/pom.xml (hunk 1 above)
Comment (j3r3m1) [reply geoindicators/pom.xml]: necessary I do not think so but it seems it optimizes the matrix manipulation:
see http://haifengl.github.io/faq.html

File: geoindicators/src/main/groovy/org/orbisgis/orbisprocess/geoclimate/geoindicators/DataUtils.groovy (hunk 3 above)
Comment (j3r3m1) [reply geoindicators/src/main/groovy/org/orbisgis/orbisprocess/geoclimate/geoindicators/DataUtils.groovy]: OK I will try to make a diff between this commit and 841781e

Comment (j3r3m1) [reply geoindicators/src/main/groovy/org/orbisgis/orbisprocess/geoclimate/geoindicators/DataUtils.groovy]: OK I did it manually but it should be OK

File: geoindicators/src/main/groovy/org/orbisgis/orbisprocess/geoclimate/geoindicators/TypologyClassification.groovy (hunk 2 above)
Comment (ebocher) [reply geoindicators/src/main/groovy/org/orbisgis/orbisprocess/geoclimate/geoindicators/TypologyClassification.groovy]: We will look on with @SPalominos  after adding dataframe supports in orbisdata

**Comment by SPalominos** [comment]
@ebocher , @j3r3m1 As a new module of orbisdata is comming with the `DataFrame` concept, maybe we can wait for it before merging this PR ?
The module will make easier the conversion of a `ITable` into a `DataFrame` and it also handle the geometries.

**Comment by ebocher** [comment]
Yes sure

**Comment by j3r3m1** [comment]
No hurry we also need to make a decision concerning the models and training data repository.
